### ISSUE TITLE
Maybe add the ability to give names to the local collections?

### ISSUE THREAD

**Issue by ivansglazunov** [dburles/mongo-collection-instances#31]
```js
abc = new Mongo.Collection(null, { localName: 'abc' });
Mongo.Collection.getLocal('abc'); // 1
Mongo.Collection.get(null, { localName: 'abc' }); // 2
Mongo.Collection.get('abc'); // 3
```
Perhaps it would be necessary to add this capability. I doubt in the interface. Which of these options is more like it? How the best option to name? `localName` somehow ugly.

**Comment by ivansglazunov** [comment]
I have the package [shuttler:ref](https://github.com/meteor-shuttler/ref) where it would be very useful.

**Comment by dburles** [comment]
Yeah that's a bit tricky, I'm not entirely sure of the potential use cases (we haven't had any issues crop up relating to looking up null collections so far)

**Comment by ivansglazunov** [comment]
so far I have 2 options:
* I will make a package `ivansglazunov:mongo-collection-instances-null` as wrapper with 3 option and `name` field.
* I will make pull request if you approve option

**Comment by dburles** [comment]
Potentially we could allow extra naming feature for both null and named collections, this may make handling multiple connections a lot more intuitive

**Comment by dburles** [comment]
Rather than passing in the connection, we can instead reference the name

**Comment by ivansglazunov** [comment]
If you specified an alternate name in the options, the main will be used?

**Comment by dburles** [comment]
```js
Foo1 = new Mongo.Collection('foo', { ref: 'foo1' });
Foo2 = new Mongo.Collection('foo', { ref: 'foo2', connection: ... });

// if a 'ref' is passed in, we use that name rather than 'foo'
Mongo.Collection.get('foo1');
Mongo.Collection.get('foo2');
```

**Comment by dburles** [comment]
I think in that case, if you define null collections without a reference, we just ignore them

**Comment by ivansglazunov** [comment]
```js
Mongo.Collection.get('foo'); // too ?
```

**Comment by ivansglazunov** [comment]
Should not be ignored. For example, we use custom adapter instead of DDP.

**Comment by dburles** [comment]
I don't think that'd be possible due to the name clash

**Comment by dburles** [comment]
Do you mean another connection?

**Comment by ivansglazunov** [comment]
Yes.
```js
new Mongo.Collection(null, { connection: ourCustomDB });
Mongo.Collection.get(null, { connection: ourCustomDB });
```

**Comment by dburles** [comment]
In that case you would also provide a ref on instantiation just like above. Will that not work in your case?

**Comment by ivansglazunov** [comment]
Yes, it was cool. Only then it is not clear for what we need the second argument of the `get`

**Comment by dburles** [comment]
Hmm, can you confirm that arbitrary options make it through to `addExtension`?

**Comment by dburles** [comment]
I think it should be alright, you will specify the `ref` option for null collections, as well as any collections that are on a different connection. Perhaps `Mongo.Collection.getRef('...')` is better?

**Comment by dburles** [comment]
That will solve this outstanding issue: https://github.com/dburles/mongo-collection-instances/issues/22

**Comment by ivansglazunov** [comment]
`get` => `getAll`
`getRef`...
Not very much.
I liked the `.get(name/ref: String)`

**Comment by dburles** [comment]
Though it would be possibly to overwrite a `ref` if we later define a collection with the same name which may be confusing?. The advantage of using `getRef`means the two are distinct.

**Comment by ivansglazunov** [comment]
It is a question of validation and comparison of patterns.
Name conflicts can not escape. If the owner of the application knows which makes, then it is a true owner of the app :)

**Comment by ivansglazunov** [comment]
Exit - namespace tree :) And I think this is not the exit.

**Comment by dburles** [comment]
Let's go with just `get`, happy for you to continue with the PR along with some tests for the new functionality

**Comment by ivansglazunov** [comment]
I think for backward compatibility should leave collection.

**Comment by dburles** [comment]
I agree, though it's not entirely working, see here: https://github.com/dburles/mongo-collection-instances/issues/22

**Comment by ivansglazunov** [comment]
May be because `connection` must be an object with `_lastSessionId` field?

**Comment by dburles** [comment]
Possibly. Let's continue discussion on the PR.
